crates/bevy_ecs/src/query/state.rs

```diff
@@ -158,8 +162,18 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
         }
     }
 
+    fn update_archetypes_unsafe_world_cell(&mut self, world: UnsafeWorldCell<'_>) {
```

Comment:
```suggestion
    pub fn update_archetypes_unsafe_world_cell(&mut self, world: UnsafeWorldCell<'_>) {
```
and also implement `update_archetypes` in terms of this method rather than the way it is now perhaps